schema/elements.go

```diff
@@ -95,6 +92,9 @@ type Struct struct {
 	// overlap between unions.
 	Unions []Union `yaml:"unions,omitempty"`
 
+	// DefaultElementType is the type of the structs's unknown fields.
```

Comment:
can it be unset? do we always allow unknown fields?

Reply:
Yes, if this is unset (set to an empty `schema.TypeRef{}`) then type resolution fails on unknown fields and so the validation still fails.